### PR TITLE
k8s append_hash

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
Add append_hash functionality to the k8s module, to provide a method for appending
hashes of ConfigMap and Secret resources to their name consistent with how kubectl achieves it.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
k8s

### CODE REVIEW

File: lib/ansible/plugins/filter/k8s.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    HAS_GENERATE_HASH = True
+except ImportError:
+    HAS_GENERATE_HASH = False
+
+from ansible.errors import AnsibleFilterError
+
+
+def k8s_config_resource_name(resource):
+    if not HAS_GENERATE_HASH:
+        raise AnsibleFilterError("k8s_config_resource_name requires openshift>=0.6.3")
```
Comment (fabianvf) [inline lib/ansible/plugins/filter/k8s.py]: Should this be `0.7.2`?

**Review by fabianvf — APPROVED** [review]

File: lib/ansible/plugins/filter/k8s.py (hunk 1 above)
Comment (willthames) [reply lib/ansible/plugins/filter/k8s.py]: Oh, yes, good catch - I've done a lot more rework on the module than the plugin, must have missed this one.

**Comment by fabianvf** [comment]
shipit

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8

**Pull request merged**